core/src/main/java/org/elasticsearch/env/Environment.java

```diff
@@ -332,4 +339,26 @@ public Path tmpFile() {
     public static FileStore getFileStore(Path path) throws IOException {
         return ESFileStore.getMatchingFileStore(path, fileStores);
     }
+
+    /**
+     * assserts that the two environments are equivalent for all things the environment cares about (i.e., all but the setting
+     * object which may contain different setting)
+     */
+    public static void assertEquivalent(Environment actual, Environment expected) {
```

Comment:
> Can you make a suggestion as to what to add?

Let's recap. I want the method moved, I don't want the public surface of `Environment` increased. You don't want to move it because you think it makes it more likely that if someone modifies `Environment`, they will modify this method. I do not agree with that, I do not think that it makes it more likely. I am saying that if you feel that keeping this method on `Environment` for the reason of making it more likely that it's modified in conjunction with `Environment` is a good enough reason to increase the public surface of `Environment`, then I want something stronger. So, to be clear, I don't even agree with the premise.

> I will try to another route and remote the Environment.getSettings() method, which means all of this goes away. I will do it in another PR

That's fine.